### PR TITLE
QS Application - Staff Task Review Updates

### PR DESCRIPTION
**Pull request by cameron-eyds**
*Issue #:*  /bcgov/entity#17244

*Description of changes:*
* Creates Qualified Supplier applicant information sub component, to use in Staff Task Review.
* Minor style adjustments and status side bar title update
* Some basic template unit testing
* Updated local env/vaults.env variables

 /bcgov/entity#17332
 /bcgov/entity#17246
 * Updated Modal content for Approval/Rejection
 * Includes additional Rejection Reason handling for Mhr Product Reviews
 
  /bcgov/entity#17247
* Updated Rejected Task Review Template to include rejection reason captured as part of flow
 
 

Screenshot 2023-09-08 at 12 34 18 PM
Screenshot 2023-09-08 at 12 34 26 PM
Screenshot 2023-09-12 at 10 23 27 AM
Screenshot 2023-09-12 at 10 23 00 AM
Screenshot 2023-09-12 at 1 32 54 PM

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2529-7kt9ox6v.web.app

File: auth-web/src/components/auth/staff/review-task/QsApplication.vue [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+  setup (props) {
+    const localState = reactive({
+      qsApplicantData: null as QualifiedSupplierApplicant,
+      qsApplicationTypeDisplay: computed(() => {
+        return props.taskDetails?.type.replace(/.*–\s*/, '')
+      }),
+      qsApplicantPhone: computed(() => {
+        return CommonUtils.toDisplayPhone(localState.qsApplicantData?.phoneNumber)
+      }),
+      qsRequirements: computed(() => {
```
Comment (seeker25) [inline auth-web/src/components/auth/staff/review-task/QsApplication.vue:203]: Cool I didn't know you could do this

Comment (cameron-eyds) [reply auth-web/src/components/auth/staff/review-task/QsApplication.vue]: Heck yea!
Wrap the local state and cut way down on the properties in the return, just spread!

File: auth-web/src/views/auth/staff/ReviewAccountView.vue [hunk 2]
```diff
@@ -399,6 +406,22 @@ export default defineComponent({
...
 
+    const componentQualifiedSupplierApplication = (tabNumber:number = 1) => {
+      return formattedComponent(
+        tabNumber,
+        `qualified-supplier-application-${tabNumber}`,
+        QsApplication,
+        {
+          title: 'Qualified Supplier Application',
+          taskDetails: task.value,
+          accountUnderReview: accountUnderReview.value,
```
Comment (seeker25) [inline auth-web/src/views/auth/staff/ReviewAccountView.vue:417]: would be cool if there was a function that expand these for you

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2529-7kt9ox6v.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2529-7kt9ox6v.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2529-7kt9ox6v.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2529-7kt9ox6v.web.app

**Comment by seeker25** [comment]
Yup no worries

**Comment by seeker25** [comment]
Is this backwards compatible? Does anything need to be hidden on release? or can we push this to prod easily? we talked and already know the secrets should be there

**Review by seeker25 — APPROVED** [review]

File: auth-web/src/assets/scss/base.scss [hunk 3]
```diff
@@ -183,4 +187,9 @@ legend {
 
 .gray-9 {
   color: $gray9;
+}
+
+.icon-text {
+  display: flex;
```
Comment (seeker25) [inline auth-web/src/assets/scss/base.scss:193]: possible to do inline? using classes?

Comment (cameron-eyds) [reply auth-web/src/assets/scss/base.scss:193]: Possibly! Will have a look

File: auth-web/src/components/auth/staff/review-task/AccountStatus.vue [hunk 4]
```diff
@@ -96,6 +104,10 @@ export default class AccountStatusTab extends Vue {
     return this.taskDetails.status === TaskStatus.HOLD
   }
 
+  private get isTaskRejected (): boolean {
+    return this.taskDetails.relationshipStatus === this.TaskRelationshipStatusEnum.REJECTED
```
Comment (seeker25) [inline auth-web/src/components/auth/staff/review-task/AccountStatus.vue:108]: If you're changing these a bit, feel free to convert to composition-api. I know you're busy working on assets stuff though haha.

**Comment by cameron-eyds** [comment]
> Is this backwards compatible? Does anything need to be hidden on release? or can we push this to prod easily? we talked and already know the secrets should be there

I believe so yes! 
This is purely for staff task review, on these QS Applications. 
I believe the task types are already visible for Staff but yea, shouldn't be anything pending for Staff to select/review.
Low impact for the most part, will require some minor regression testing on the Dialogs/Modals however to current flows.

File: auth-web/src/components/auth/staff/review-task/AccountStatus.vue (hunk 4 above)
Comment (cameron-eyds) [reply auth-web/src/components/auth/staff/review-task/AccountStatus.vue:108]: I'll have a look, i don't mind!

**Review by ochiu — APPROVED** [review]

**Comment by seeker25** [comment]
as discussed - Divya will QA this ticket

**Pull request merged**